internal/k8s/helpers.go

```diff
@@ -83,3 +87,60 @@ func isStatusEqual(objA, objB interface{}) bool {
 	}
 	return false
 }
+
+// IsObjectEqual checks if two objects are equal.
+//
+// Make an attempt to avoid comparing full objects since it is very CPU intensive.
+// Prefer comparing Generation when it is implemented in Kubernetes, e.g. all CRDs implement it.
+// Otherwise compare the ResourceVersion or compare the full object.
+func IsObjectEqual(a, b interface{}) bool {
+	switch a.(type) {
+	case *v1.Secret:
+		return isResourceVersionEqual(a, b)
+	case *v1.Service:
+		return isResourceVersionEqual(a, b)
+	case *v1.Endpoints:
+		return isResourceVersionEqual(a, b)
+
+	case *networking_v1.Ingress:
+		return isGenerationEqual(a, b)
+	case *contour_api_v1.HTTPProxy:
+		return isGenerationEqual(a, b)
+	case *contour_api_v1alpha1.ExtensionService:
+		return isGenerationEqual(a, b)
+	case *contour_api_v1.TLSCertificateDelegation:
+		return isGenerationEqual(a, b)
+	case *contour_api_v1alpha1.ContourConfiguration:
+		return isGenerationEqual(a, b)
+
+	case *gatewayapi_v1beta1.GatewayClass:
+		return isGenerationEqual(a, b)
+	case *gatewayapi_v1beta1.Gateway:
+		return isGenerationEqual(a, b)
+	case *gatewayapi_v1beta1.HTTPRoute:
+		return isGenerationEqual(a, b)
+	case *gatewayapi_v1alpha2.TLSRoute:
+		return isGenerationEqual(a, b)
+
+	// Fallback implementation: compare the full object if not handled above.
```

Comment:
I'm thinking we could probably just use ResourceVersion comparison as the fallback strategy for any resources that Contour is not updating the status of, WDYT? Would be nice and simple. I don't really see a need to deep-compare.

Reply:
Yes, I agree.

Reply:
> I'm thinking we could probably just use ResourceVersion comparison as the fallback strategy for any resources that Contour is not updating the status of, WDYT? Would be nice and simple. I don't really see a need to deep-compare.

Since it is just fallback, it would be OK even though [`ManagedFields`](https://kubernetes.io/docs/reference/using-api/server-side-apply/#field-management) had caused performance issues previously https://github.com/projectcontour/contour/pull/2987?